### PR TITLE
[App Service] `az webapp up/create/update`: Add new parameter `--basic-auth` to allow users to enable and disable basic auth

### PR DESCRIPTION
**Pull request by shnjefff**
**Related command**

az webapp up -g <rg-name> --name <app-name> --basic-auth enabled/disabled
az webapp update -g <rg-name> --name <app-name> --basic-auth enabled/disabled
az webapp create -g <rg-name> -p <serverfarm-name> --name <app-name> --basic-auth enabled/disabled

**Description**

This pr add --basic-auth parameter to webapp create/up/update so that customers can toggle basic auth settings. The default value for --basic-auth is "" (empty string), which will not change existing settings. Customers only allow to input two values: "enabled" and "disabled" case insensitively. Any other values will raise exception. 

**Testing Guide**

create a webapp and use related command to change the basic auth setting
If enabled, you should see basic auth publishing as on. Otherwise off
image

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
App Service

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py [hunk 1]
```diff
@@ -160,6 +160,7 @@ def load_arguments(self, _):
         c.argument('subnet', help="Name or resource ID of the pre-existing subnet to have the webapp join. The --vnet is argument also needed if specifying subnet by name.")
         c.argument('public_network_access', help="Enable or disable public access to the web app", arg_type=get_enum_type(PUBLIC_NETWORK_ACCESS_MODES))
         c.argument('acr_use_identity', action='store_true', help="Enable or disable pull image from acr use managed identity")
+        c.argument('basic_auth', options_list=['--basic-auth'], help='Enable or disable basic auth.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/appservice/_params.py:163]:
```suggestion
        c.argument('basic_auth', action='store_true', help='Enable or disable basic auth.')
```

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py [hunk 2]
```diff
@@ -209,6 +210,7 @@ def load_arguments(self, _):
                    arg_type=get_three_state_flag(return_label=True), deprecate_info=c.deprecate(expiration='3.0.0'))
         c.argument('minimum_elastic_instance_count', options_list=["--minimum-elastic-instance-count", "-i"], type=int, is_preview=True, help="Minimum number of instances. App must be in an elastic scale App Service Plan.")
         c.argument('prewarmed_instance_count', options_list=["--prewarmed-instance-count", "-w"], type=int, is_preview=True, help="Number of preWarmed instances. App must be in an elastic scale App Service Plan.")
+        c.argument('basic_auth', options_list=['--basic-auth'], help='Enable or disable basic auth.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/appservice/_params.py:213]:
```suggestion
        c.argument('basic_auth', action='store_true', help='Enable or disable basic auth.')
```

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py [hunk 3]
```diff
@@ -689,6 +691,7 @@ def load_arguments(self, _):
                    default=False, action='store_true')
         c.argument('html', help="Ignore app detection and deploy as an html app", default=False, action='store_true')
         c.argument('app_service_environment', options_list=['--app-service-environment', '-e'], help='name or resource ID of the (pre-existing) App Service Environment to deploy to. Requires an Isolated V2 sku [I1v2, I2v2, I3v2]')
+        c.argument('basic_auth', options_list=['--basic-auth'], help='Enable or disable basic auth.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/appservice/_params.py:694]:
```suggestion
        c.argument('basic_auth', action='store_true', help='Enable or disable basic auth.')
```

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py (hunk 1 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/appservice/_params.py:163]: > options_list=['--basic-auth']

There is no need to explicitly declare the default option name

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/appservice/_params.py:163]: >  action='store_true'

Please select the appropriate type for the Bool variable

**Comment by zhoxing-ms** [comment]
Please add some scenario tests for this new feature

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py [hunk 4]
```diff
@@ -160,6 +160,7 @@ def load_arguments(self, _):
         c.argument('subnet', help="Name or resource ID of the pre-existing subnet to have the webapp join. The --vnet is argument also needed if specifying subnet by name.")
         c.argument('public_network_access', help="Enable or disable public access to the web app", arg_type=get_enum_type(PUBLIC_NETWORK_ACCESS_MODES))
         c.argument('acr_use_identity', action='store_true', help="Enable or disable pull image from acr use managed identity")
+        c.argument('basic_auth', type=str, help='Enable or disable basic auth.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/appservice/_params.py:163]: Please refer to `public_network_access` parameter and define it as `get_enum_type`

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**